controller/contour/controller.go

```diff
@@ -103,26 +103,47 @@ func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
 
 // ensureContour ensures all necessary resources exist for the given contour
 // in namespace ns.
-func (r *Reconciler) ensureContour(ctx context.Context, contour *operatorv1alpha1.Contour, ns string) error {
+func (r *Reconciler) ensureContour(ctx context.Context, contour *operatorv1alpha1.Contour) error {
 	// Before doing anything with the contour, ensure it has a finalizer
 	// so it can cleaned-up later.
 	if err := r.ensureFinalizer(ctx, contour); err != nil {
 		return fmt.Errorf("failed to finalize contour %s/%s: %w", contour.Namespace, contour.Name, err)
 	}
-	if err := r.ensureNamespace(ctx, ns); err != nil {
-		return fmt.Errorf("failed to ensure namespace %s: %w", ns, err)
+	if err := r.ensureNamespace(ctx, contour); err != nil {
+		return fmt.Errorf("failed to ensure namespace %s for contour %s/%s: %w",
+			contour.Spec.Namespace.Name, contour.Namespace, contour.Name, err)
 	}
 	return nil
 }
 
 // ensureContourRemoved ensures all resources for the given contour do not exist
 // in namespace ns.
-func (r *Reconciler) ensureContourRemoved(ctx context.Context, contour *operatorv1alpha1.Contour, ns string) error {
-	if err := r.ensureNamespaceRemoved(ctx, defaultContourNamespace); err != nil {
-		return fmt.Errorf("failed to remove namespace %s: %w", defaultContourNamespace, err)
+func (r *Reconciler) ensureContourRemoved(ctx context.Context, contour *operatorv1alpha1.Contour) error {
+	if err := r.ensureNamespaceRemoved(ctx, contour); err != nil {
+		return fmt.Errorf("failed to remove namespace %s for contour %s/%s: %w",
+			contour.Spec.Namespace.Name, contour.Namespace, contour.Name, err)
 	}
 	if err := r.ensureFinalizerRemoved(ctx, contour); err != nil {
 		return fmt.Errorf("failed to remove finalizer from contour %s/%s: %w", contour.Namespace, contour.Name, err)
 	}
 	return nil
 }
+
+// contoursExist lists Contour objects in all namespaces if all is true or only the
+// namespace of contour otherwise, returning true if any exist other than contour.
+func (r *Reconciler) contoursExist(ctx context.Context, contour *operatorv1alpha1.Contour, all bool) (bool, error) {
```

Comment:
`all` is not currently specified as `true` anywhere, but I assume you have a plan to use it?

Reply:
I replaced this func with https://github.com/projectcontour/contour-operator/pull/21/files#diff-eee6f38bbe0411df081f7393625f732eR136-R159. I'll remove either based on which PR merges first.